### PR TITLE
[exporter/azuremonitor] Refactor to use higher level api to send trace events

### PR DESCRIPTION
**Pull request by whitneygriffith**
#### Description

1. Created the spanToTelemetry method to leverage the higher-level Track API, which is the recommended approach for sending traces to the queue. This refactor simplifies trace processing and aligns with best practices for handling telemetry data efficiently.
2. Resolved an issue where traces weren't being sent to App Insights due to not flushing the Telemetry Client. Added the necessary flush operation to ensure all traces are properly sent to the queue, leveraging App Insights' batch handling for more efficient processing.

#### Link to tracking issue
Part of #35037 

I will follow up with updating the API used for sending metrics and logs as well. When this is done I will deprecate the `transportChannel` which uses the lower level `Send()` method

### CODE REVIEW

File: exporter/azuremonitorexporter/trace_to_envelope.go [hunk 1]
```diff
@@ -40,6 +41,106 @@ var (
...
+
+	// Add instrumentation scope details
+	trace.Properties["instrumentation.library.name"] = instrumentationScope.Name()
+	trace.Properties["instrumentation.library.version"] = instrumentationScope.Version()
+
+	// Add the trace telemetry to the list
+	traces = append(traces, trace)
+
+	// Handle span events (these are OpenTelemetry events within a span)
+	for i := 0; i < span.Events().Len(); i++ {
```
Comment (whitneygriffith) [inline exporter/azuremonitorexporter/trace_to_envelope.go:105]: Update to for each

File: exporter/azuremonitorexporter/trace_to_envelope.go [hunk 2]
```diff
@@ -40,6 +41,106 @@ var (
...
+		spanEvent := span.Events().At(i)
+
+		// Skip non-exception events if configured
+		if spanEvent.Name() != exceptionSpanEventName && !spanEventsEnabled {
```
Comment (keithmattix) [inline exporter/azuremonitorexporter/trace_to_envelope.go:109]: nit(readability): Putting `spanEventsEnabled` as the first expression is a bit easier to read

File: exporter/azuremonitorexporter/trace_to_envelope.go (hunk 1 above)
Comment (keithmattix) [inline exporter/azuremonitorexporter/trace_to_envelope.go:105]: `for _, spanEvent := span.Events()` would be a bit more idiomatic

Comment (whitneygriffith) [reply exporter/azuremonitorexporter/trace_to_envelope.go:105]: cannot range over `span.Events()` value of type `ptrace.SpanEventSlice` as it does not support ranging. The `.At()` method must be used to access each event in a traditional for loop

Comment (whitneygriffith) [reply exporter/azuremonitorexporter/trace_to_envelope.go:105]: cannot range over `span.Events()` value of type `ptrace.SpanEventSlice` as it does not support ranging. The `.At()` method must be used to access each event in a traditional for loop

**Pull request closed without merging**